packages/components/src/components/entities/actions.ts

```diff
@@ -337,3 +340,39 @@ export function deleteEntityType(deleteActionName: string, rowId: number): Promi
         });
     });
 }
+
+/**
+ * Delete a sample or selection of samples from a given sample type (based on the QueryGridModel).
+ * @param model - QueryGridModel with row data for the sample type schema/query
+ * @param rows - the rows that are to be deleted.  Each element of the array contains, minimally,  RowId: <value>
+ * @param nounSingular - singular noun for the type being deleted
+ * @param nounPlural - plural noun for the type being deleted
+ * @param onComplete - callback function for after the deleteRows command finishes
+ * @param onFailure? - callback function for failure to delete samples
+ * @param auditBehavior? - optional property to indicate the audit behavior for this query
+ */
+export function deleteEntityDataRows(
```

Comment:
I realize this was factored out, but this function doesn't seem to provide much value over and above direct calls to `deleteRows`. I think each caller could do the mapping of the `SchemaQuery` themselves and get the same value without needing this method.

Reply:
I agree with Nick's comment. When this function was moved out of SM, it included the on success and on failure behaviors so it made sense as a separate function in that case to allow for those to be consistent between usages. But porting this to ui-components and having the caller pass in the onComplete and onFailure means that they could likely just call `deleteRows` directly just as easily.

Reply:
Good call.  Fixed.  Calling deleteRows directly